src/terms/xc.jl

```diff
@@ -451,38 +457,164 @@ end
 function Libxc.evaluate(xc::Functional, density::LibxcDensities;
                         derivatives=0:1, skip_unsupported_derivatives=false, kwargs...)
     if skip_unsupported_derivatives
-        derivatives = filter(i -> i in supported_derivatives(xc), derivatives)
+        derivatives = filter(i -> i in Libxc.supported_derivatives(xc), derivatives)
     end
+    zk, vrho, vsigma, v2rho2 = nothing, nothing, nothing, nothing
     if xc.family == :lda
-        evaluate(xc; rho=density.ρ_real, derivatives, kwargs...)
+        #@warn "xc lda evaluate"
+        zk, vrho, vsigma, v2rho2 = evaluate(xc; rho=density.ρ_real, derivatives, kwargs...)
     elseif xc.family == :gga
-        evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, derivatives, kwargs...)
+        #@warn "xc gga evaluate"
+        zk, vrho, vsigma, v2rho2 = evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, derivatives, kwargs...)
     elseif xc.family == :mgga && !needs_laplacian(xc)
-        evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, tau=density.τ_real,
+        #@warn "xc mgga evaluate"
+        zk, vrho, vsigma, v2rho2 = evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, tau=density.τ_real,
                  derivatives, kwargs...)
     elseif xc.family == :mgga && needs_laplacian(xc)
-        evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, tau=density.τ_real,
+        #@warn "xc mgga evaluate"
+        zk, vrho, vsigma, v2rho2 = evaluate(xc; rho=density.ρ_real, sigma=density.σ_real, tau=density.τ_real,
                  lapl=density.Δρ_real, derivatives, kwargs...)
     else
         error("Not implemented for functional familiy $(xc.family)")
     end
+    #return (; zk, vrho, v2rho2, vsigma)
+    return zk, vrho, vsigma, v2rho2
 end
+
 function Libxc.evaluate(xcs::Vector{Functional}, density::LibxcDensities; kwargs...)
     isempty(xcs) && return NamedTuple()
-    result = evaluate(xcs[1], density; kwargs...)
-    for i in 2:length(xcs)
-        other = evaluate(xcs[i], density; kwargs...)
-        for (key, data) in pairs(other)
-            if haskey(result, key)
-                result[key] .+= data
-            else
-                result = merge(result, (key => data, ))
+    # conversion from NamedTuple and back can hopefully be skipped once mergewith is
+    # implemented for NamedTuple in Julia (open PR)
```

Comment:
I have the same issue in my PR, but I think for now it is fair to hard-code it for the case of just two functionals. The set of keys even for more general cases (beyond LDA) is large, but manageable, so one could perhaps unroll everything manually using some code generation tricks. But for sure this is messy.

I think the advantage with my PR is that you can specialise on `Vector{<:FallbackFunctional}` and thereafter assume that the returned `terms` may at most have `zk`, `vrho`, `vsigma` and `v2rho2`. That could help.